velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -340,7 +350,19 @@ bool registerAverageAggregate(const std::string& name) {
           checkSumCountRowType(
               inputType,
               "Input type for final aggregation must be (sum:double, count:bigint) struct");
-          return std::make_unique<AverageAggregate<int64_t>>(resultType);
+          switch (resultType->kind()) {
+            case TypeKind::REAL:
+              return std::make_unique<AverageAggregate<int64_t, double, float>>(
+                  resultType);
+            case TypeKind::DOUBLE:
+            case TypeKind::ROW:
```

Comment:
Could this return type of this function be ROW?

Reply:
Yes, it can be for aggregation of intermediate to intermediate types. Here are the possible steps of aggregations and their inputs and outputs:



Step | Input | Output
-- | -- | --
Partial | Raw Data | Intermediate Results
Final | Intermediate Results | Final Results
Single | Raw Data | Final Results
Intermediate | Intermediate Results | Intermediate Results

Refer: https://facebookincubator.github.io/velox/develop/aggregate-functions.html

Reply:
Ah, makes sense. Thanks for the explanation!